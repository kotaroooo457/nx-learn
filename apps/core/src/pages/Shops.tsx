import { useRecoilValue } from 'recoil';

import { shopListState } from '../store/shopsSelector';

export const Shops: React.FC = () => {
  const shops = useRecoilValue(shopListState);

  return (
    <div>
      <div>Shops</div>
      <ul className="flex justify-center">
        {shops.map(({ id, name }) => {
          return <li key={id}>{name}</li>;
        })}
      </ul>
    </div>
  );
};
