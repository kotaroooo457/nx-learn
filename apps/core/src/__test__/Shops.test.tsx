import { render, screen, waitFor } from '@testing-library/react';
import { Suspense } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { RecoilRoot } from 'recoil';

import { Shops } from '../pages/Shops';

describe('<Shops />', () => {
  test('正常にレンダリングできる', async () => {
    const { container } = render(
      <RecoilRoot>
        <BrowserRouter>
          <Suspense fallback="...loading">
            <Shops />
          </Suspense>
        </BrowserRouter>
      </RecoilRoot>
    );

    await waitFor(() => expect(screen.getByText('ドラッグストア')).toBeTruthy());

    expect(container).toMatchSnapshot();
  });
});
