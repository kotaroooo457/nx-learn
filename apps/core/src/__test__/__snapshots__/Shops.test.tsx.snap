// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shops /> 正常にレンダリングできる 1`] = `
<div>
  <div>
    <div>
      Shops
    </div>
    <ul
      class="flex justify-center"
    >
      <li>
        ドラッグストア
      </li>
      <li>
        サンドラッグ
      </li>
      <li>
        マツモトキヨシ
      </li>
    </ul>
  </div>
</div>
`;
